reactor-netty-http/src/main/java/reactor/netty/http/server/logging/AccessLogArgProviderH2.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2011-Present VMware, Inc. or its affiliates, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.netty.http.server.logging;
+
+import io.netty.handler.codec.http2.Http2HeadersFrame;
+
+import java.net.SocketAddress;
+import java.util.Objects;
+
+/**
+ * @author limaoning
+ */
+final class AccessLogArgProviderH2 extends AbstractAccessLogArgProvider<AccessLogArgProviderH2> {
+
+	static final String H2_PROTOCOL_NAME = "HTTP/2.0";
+
+	Http2HeadersFrame requestHeaders;
+
+	AccessLogArgProviderH2(SocketAddress remoteAddress) {
+		super(remoteAddress);
+	}
+
+	AccessLogArgProviderH2 requestHeaders(Http2HeadersFrame requestHeaders) {
+		this.requestHeaders = Objects.requireNonNull(requestHeaders, "requestHeaders");
+		onRequest();
+		return self();
+	}
+
+	@Override
+	public CharSequence method() {
+		return requestHeaders == null ? null : requestHeaders.headers().method();
+	}
+
+	@Override
+	public CharSequence uri() {
+		return requestHeaders == null ? null : requestHeaders.headers().path();
+	}
+
+	@Override
+	public String protocol() {
+		return H2_PROTOCOL_NAME;
+	}
```

Comment:
Can we move those in the abstract class and then in the children to override the onRequest method where we will initialise method, uri and protocol